### PR TITLE
Add Associations between classes

### PR DESCRIPTION
**Pull request by jrosario19**
### In this PR I added the following feature:
- Create a class Classroom with the following:
  - @label instance variable, should be initialized in the constructor.
  - Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
  - Create the has-many side (a classroom has many students).
  - Create the belongs-to side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
  - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -5,14 +5,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (omar-labana) [inline person.rb:8]: - [ ] The `Student` class should have a method that serves as the 'belongs-to' side (a student belongs to a classroom).

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
```
Comment (omar-labana) [inline book.rb:2]: - [ ] A method for adding rentals should be added to the Book class, please review [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars).

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @jrosario19  

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Whoistolu) [reply person.rb:8]: Well resolved 👍🏽

File: book.rb (hunk 2 above)
Comment (Whoistolu) [reply book.rb:2]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @jrosario19 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**